gaia-sdk-java/gaia-sdk-http/src/main/kotlin/gaia/sdk/http/HttpTransporter.kt

```diff
@@ -34,41 +30,48 @@ class HttpTransporter(private val url: String, private val httpClient: HttpClien
             log.debug("Payload to send: '${String(bytes)}'")
         }
 
-        return httpClient.headers {
-            it.add("Content-Type", options.contentType)
-            it.add("Authorization", buildAuthorizationHeader(options, String(bytes)))
-        }
+        return httpClient
+                .headers {
+                    it.add("Content-Type", options.contentType)
+                    it.add("Authorization", buildAuthorizationHeader(options, String(bytes)))
+                }
                 .followRedirect(true)
                 .post()
                 .uri(url)
                 .send(Mono.just(Unpooled.copiedBuffer(bytes)))
                 .responseConnection { t, u ->
-                    if (t.status().code() >= 400) {
-                        val tuple = Tuples.of(t.status(), ByteArray(0))
-                        return@responseConnection Flux.just(tuple)
-                    }
-                    val publisher1: Flux<HttpResponseStatus> = Flux.just(t.status())
-                    val publisher2: Flux<ByteArray> = u.inbound().receive().asByteArray().buffer().map { list ->
-                        val bos = ByteArrayOutputStream()
-                        list.forEach(bos::write)
-                        bos.toByteArray()
-                    }
+                    u
+                        .inbound()
+                        .receive()
+                        .asByteArray()
+                        .buffer()
+                        .map { list ->
+                            val bos = ByteArrayOutputStream()
+                            list.forEach(bos::write)
+                            bos.toByteArray()
+                        }
+                        .switchIfEmpty(
+                            Flux.just(t.status())
+                                .flatMap {
+                                    if (it.code() >= 400) {
+                                        val msg = "Error with status code ${t.status().code()} (${t.status().reasonPhrase()}) and no payload"
+                                        Flux.error(HttpTransportException(msg))
+                                    } else {
+                                        Flux.empty<ByteArray>()
+                                    }
+                                }
+                        )
+                        .map { byteArray ->
+                            if (t.status().code() >= 400) {
+                                val msg = "Error with status code ${t.status().code()} (${t.status().reasonPhrase()}) and payload: ${String(byteArray)}"
```

Comment:
should we limit the payload that is included in the error message?
if so, what limit?

Reply:
imo not necessary